### ISSUE TITLE
Using the <explanation> placeholder in a suppression causes it not to work

### ISSUE THREAD

**Issue by GiuseppeIII** [biomejs/biome#5846]
### Environment information

```bash
...
```

### Rule name

All

### Playground link

https://next.biomejs.dev/playground/?lintRules=all&code=LwAqACoAIABiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAtAGEAbABsACAAbABpAG4AdAAvAG4AdQByAHMAZQByAHkALwBuAG8AUAByAG8AYwBlAHMAcwBFAG4AdgA6ACAAPABlAHgAcABsAGEAbgBhAHQAaQBvAG4APgAgACoALwAKAC8AKgAqACAAYgBpAG8AbQBlAC0AaQBnAG4AbwByAGUALQBhAGwAbAAgAGwAaQBuAHQALwBuAHUAcgBzAGUAcgB5AC8AbgBvAFAAcgBvAGMAZQBzAHMARwBsAG8AYgBhAGwAOgAgADwAZQB4AHAAbABhAG4AYQB0AGkAbwBuAD4AIAAqAC8ACgAKAHAAcgBvAGMAZQBzAHMALgBlAG4AdgA%3D

### Expected result

Using the <explanation> placeholder in a suppression causes it not to work. (See playground link). This is new behavior from beta.1. Ideally having the suppression at all should actually suppress as there is already a seperate `A suppression shouldn't have a <explanation> placeholder.` waring anyway.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
This should be easy to fix. We need to check the severity of the diagnostic, and it's below error, we need to emit a diagnostic but continuing to process the suppression.


This could be a good first issue

**Comment by ematipico** [comment]
Here's the code that should be modified 

https://github.com/biomejs/biome/blob/main/crates%2Fbiome_js_analyze%2Fsrc%2Flib.rs#L94-L101

Of course, the fix must be applied to every language

**Comment by ematipico** [comment]
I am rereading this issue, and I believe I didn't understand the problem. The suppression is *broken*, which means that it can't suppress the rule.

Hey @siketyan, you initially triaged the issue, what's the problem here?

**Comment by siketyan** [comment]
@ematipico As the emitted diagnostics are **warning**, I think we can continue using the suppression comment. Though the placeholder `<explanation>` is not intended to be kept as-is, it _is_ a valid suppression syntax. If we want to reject these suppressions completely, we should emit diagnostics with `error` severity.

**Comment by ematipico** [comment]
Ah, I see. Then that's not beginner friendly :) I will work on it
